src/project/project-context.ts

```diff
@@ -477,6 +479,7 @@ export async function projectContext(
             temp,
             cleanup: () => {
               context.diskCache.close();
+              temp.cleanup();
```

Comment:
This cleanup was for singleFile render associated to `const temp = globalTempContext();`
So as this is already cleaned up globally, I didn't think we needed to do manual temp.cleanup in this case. 

TBH, having those two different tempContext is for now not easy for me to reason about. I guess cleaning up in this finally step will not prevent or mess the last cleanup 

1. Global cleanup register for `quarto-session` dir in TEMP
https://github.com/quarto-dev/quarto-cli/blob/6164de81bf04ffab974e912ce424ebb04cfb6cfd/src/quarto.ts#L185

2. Called in Exit cleanup
https://github.com/quarto-dev/quarto-cli/blob/6164de81bf04ffab974e912ce424ebb04cfb6cfd/src/core/cleanup.ts#L11-L26

I guess this will be unused code now if we do cleanup as part of `context.cleanup`

Reply:
> TBH, having those two different tempContext is for now not easy for me to reason about.

The old code is (demonstrably, see this PR!) hard for me to reason about too...

Reply:
And actually since this is also not a single-file render, I think we should replace globalTempContext() with a makeTempContext like we do in the other settings.